### PR TITLE
Extend DbtTask with additional supported fields and document them

### PR DESCRIPTION
**Pull request by gaborratky-db**
## What changes are proposed in this pull request?

* Add additional fields that the upcoming dbt task type [1] supports:
  * `ProfilesDirectory` - Optional field to configure where `dbt` should look for `profiles.yml` within the repository.
  * `WarehouseId` - Optional field configuring a Databricks SQL Warehouse that `dbt` should run the queries against.
* Allow configuring a job with a single dbt task
* Document `dbt_task` in docs/resources/job.md.

## How is this tested?

E2E tested with https://github.com/gaborratky-db/jaffle_shop/tree/main/terrafood

[1] https://databricks.com/blog/2022/06/29/top-5-workflows-announcements-at-data-ai-summit.html

### CODE REVIEW

File: docs/resources/job.md [hunk 1]
```diff
@@ -177,6 +177,16 @@ You can invoke Spark submit tasks only on new clusters. **In the `new_cluster` s
 * `parameters` - (Optional) Parameters for the task
 * `named_parameters` - (Optional) Named parameters for the task
 
+### dbt_task Configuration Block
+
+* `commands` - (Required) (Array) Series of dbt commands to execute in sequence. Every command must start with "dbt".
+* `project_directory` - (Optional) The relative path to the directory in the repository where dbt should look in for the `dbt_project.yml` file. If not specified, defaults to the repository's root directory. Equivalent to passing `--project-dir` to a dbt command.
```
Comment (nfx) [inline docs/resources/job.md:183]:
```suggestion
* `project_directory` - (Optional) The relative path to the directory in the repository specified in `git_source` where dbt should look in for the `dbt_project.yml` file. If not specified, defaults to the repository's root directory. Equivalent to passing `--project-dir` to a dbt command.
```

File: docs/resources/job.md [hunk 2]
```diff
@@ -177,6 +177,16 @@ You can invoke Spark submit tasks only on new clusters. **In the `new_cluster` s
...
+* `profiles_directory` - (Optional) The relative path to the directory in the repository where dbt should look in for the `profiles.yml` file. If not specified, defaults to the repository's root directory. Equivalent to passing `--profile-dir` to a dbt command.
```
Comment (nfx) [inline docs/resources/job.md:184]:
```suggestion
* `profiles_directory` - (Optional) The relative path to the directory in the repository specified by `git_source` where dbt should look in for the `profiles.yml` file. If not specified, defaults to the repository's root directory. Equivalent to passing `--profile-dir` to a dbt command.
```

**Review by nfx — APPROVED** [review]
👍

**Comment by nfx** [comment]
@gaborratky-db is this functionality working now? :)

**Comment by gaborratky-db** [comment]
> @gaborratky-db is this functionality working now? :)

@nfx, yes, dbt tasks are available for all Premium and Enterprise Databricks customers.

**Pull request merged**